### PR TITLE
fix: projects filters

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: regen-network/rnd-dev-team#1854

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

Go to https://deploy-preview-2286--regen-marketplace.netlify.app/projects/1 and test the updated credit class filters:
- select/clear all
- unregistered projects

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by blushi** [comment]
@erikalogie see testing instructions

File: web-marketplace/src/hooks/projects/useProjectsWithOrders.ts [hunk 1]
```diff
@@ -241,6 +247,8 @@ export function useProjectsWithOrders({
   });
 
   return {
+    allProjects: projectsWithOrderData,
+    haveOffChainProjects: allOffChainProjects?.length > 0,
```
Comment (blushi) [inline web-marketplace/src/hooks/projects/useProjectsWithOrders.ts:251]: we need this now in order to display in the projects filter only the credit classes that have projects and the "unregistered projects" filtered only if there are approved off chain projects

File: web-marketplace/src/pages/Projects/Projects.SideFilter.tsx [hunk 2]
```diff
@@ -127,15 +179,16 @@ export const ProjectsSideFilter = ({
...
-                              track<
-                                'filterCreditClass',
-                                FilterCreditClassEvent
-                              >('filterCreditClass', {
-                                creditClassId: path,
-                                selected: getFilterSelected(
-                                  event.target.checked,
-                                ),
-                              });
+                              if (path !== UNREGISTERED_PATH)
```
Comment (blushi) [inline web-marketplace/src/pages/Projects/Projects.SideFilter.tsx:182]: maybe we want to start tracking users clicking on "unregistered projects" filter, although I don't think we had that for the old "show off chain projects" filter cc/ @erikalogie

File: web-marketplace/src/pages/Projects/Projects.normalizers.ts [hunk 3]
```diff
@@ -1,40 +1,58 @@
...
+  const creditClassesIdsWithProjects = [
+    ...new Set(allProjects.map(project => project.creditClassId)),
+  ];
+
+  const creditClassFilters: CreditClassFilter[] =
+    creditClassesWithMetadata
+      ?.filter(
+        ({ creditClass }) =>
+          creditClass.id !== SKIPPED_CLASS_ID &&
+          creditClassesIdsWithProjects.includes(creditClass.id),
```
Comment (blushi) [inline web-marketplace/src/pages/Projects/Projects.normalizers.ts:38]: filtering credit classes without any projects so they don't show up in the filters side bar

**Comment by erikalogie** [comment]
LGTM!!!

**Comment by aaronc** [comment]
I don't see an unregistered projects filter option. Instead it says "Name from Metadata". 

image

**Comment by blushi** [comment]
"Name from metadata" is just the name of some testing credit class
I assume you don't see the "unregistered projects" filter because of this cross domain cookie safari settings (we only display this filter if the server responds with some off chain projects, and this query fails in this case because of the csrf cookie that is not passed)
Please disable this setting from safari or use another browser for testing.

**Review by aaronc — APPROVED** [review]
Works for me now.

@erikalogie unregistered projects is fine for now but maybe in the future we can find a better name like "Early Stage Projects"? Or we figure out some more general story around discovery of unregistered projects.

**Comment by erikalogie** [comment]
> Works for me now.
> 
> @erikalogie unregistered projects is fine for now but maybe in the future we can find a better name like "Early Stage Projects"? Or we figure out some more general story around discovery of unregistered projects.

I like "Early-stage projects", wdyt @clevinson?

**Pull request merged**